src/components/Envoy/EnvoyModal.tsx

```diff
@@ -42,7 +46,11 @@ const toolbarSpace = style({
   marginLeft: '1em'
 });
 
-type EnvoyDetailProps = {
+type ReduxProps = {
+  namespaces: Namespace[] | undefined;
```

Comment:
I don't think "undefined" is necessary.
When you connect the component with Redux to sync with namespaces you'll get always a value (or an empty array).

At least other components that are connected to the "namespaces" prop doesn't add the undefined.